### PR TITLE
Implemented multi-threading for live test

### PR DESCRIPTION
**Pull request by vidai-msft**
## Description

The main purpose for this PR is to implement multi-threading which leverages PowerShell runspace pool.
In addition, it also has following update for live test.
- Re-design the logic for email notification. Instead of retrieving the errors from Kusto again, now it retrieves the errors from the raw csv file. This change is due to a recent update that only scheduled live test failures will be saved to Kusto. If still with the old logic, then the errors cannot be obtained.
- Use auto mapped environment variables in Azure pipeline to shorten the list of parameters in each PowerShell task script.
- Replace the Write-Host with Write-Output in live test framework. This is due to Write-Output is well organized in runspace pool while Write-Host would make the output in Azure pipeline like spaghetti

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: tools/TestFx/Live/InvokeLiveTestScenarios.ps1 [hunk 1]
```diff
@@ -1,66 +1,206 @@
...
+$liveScenarios = Get-ChildItem -Path $srcDir -Directory -Exclude "Accounts" -ErrorAction SilentlyContinue | Get-ChildItem -Directory -Filter "LiveTests" -Recurse | Get-ChildItem -File -Filter "TestLiveScenarios.ps1" | Select-Object -ExpandProperty FullName
 
-    [Parameter(Mandatory)]
-    [ValidateScript({ Test-Path -LiteralPath $_ -PathType Container })]
-    [string] $DataLocation
-)
+$maxRunspaces = 9
+[void][int]::TryParse(${env:RSPTHROTTLE}, [ref]$maxRunspaces)
+$rsp = [runspacefactory]::CreateRunspacePool(1, $maxRunspaces)
+$rsp.CleanupInterval = [timespan]::FromHours(10)
```
Comment (isra-fel) [inline tools/TestFx/Live/InvokeLiveTestScenarios.ps1:21]: // runspace may be disposed when waiting for resources to be removed. default 15min

File: tools/TestFx/Live/SaveLiveTestResult.ps1 [hunk 2]
```diff
@@ -17,60 +17,15 @@ param (
...
-        @{ Name = "ExtendedProperties"; Expression = { $exProps } } |
-        Export-Csv -Path $_ -Encoding utf8 -NoTypeInformation -Force
-    }
+$ltDir = Join-Path -Path ${env:DATALOCATION} -ChildPath "LiveTestAnalysis" | Join-Path -ChildPath "Raw"
+$ltResults = Get-ChildItem -Path $ltDir -Filter "*.csv" -File -ErrorAction SilentlyContinue | Select-Object -ExpandProperty FullName
 
+if (![string]::IsNullOrEmpty($ltResults)) {
     Import-Module "./tools/TestFx/Utilities/KustoUtility.psd1" -ArgumentList $ServicePrincipalTenantId, $ServicePrincipalId, $ServicePrincipalSecret, $ClusterName, $ClusterRegion -Force
-    Import-KustoDataFromCsv -DatabaseName $DatabaseName -TableName $TableName -CsvFile $liveTestResults
+    Import-KustoDataFromCsv -DatabaseName ${env:LIVETESTDATABASENAME} -TableName ${env:LIVETESTTABLENAME} -CsvFile $ltResults
```
Comment (isra-fel) [inline tools/TestFx/Live/SaveLiveTestResult.ps1:28]: Import is a little confusing. consider publish

File: .azure-pipelines/live-test.yml [hunk 3]
```diff
@@ -43,6 +43,7 @@ parameters:
 variables:
   ArtifactsName: LiveTestArtifacts
   DataLocation: $(Pipeline.Workspace)/$(ArtifactsName)
+  SuppressAzurePowerShellBreakingChangeWarnings: true
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply .azure-pipelines/live-test.yml:46]: This variable is used to suppress breaking change messages in Azure PowerShell

File: tools/TestFx/Live/SendLiveTestReport.ps1 [hunk 4]
```diff
@@ -79,19 +50,46 @@ $css = @"
...
+        width: 50%;
     }
     </style>
 "@
 
-$emailSubject = "Live Test Status Report"
+$emailSubject = "Azure PowerShell Live Test Status Report"
+
+$summarySection = @"
+    <h1>Live Test Summary</h1>
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply tools/TestFx/Live/SendLiveTestReport.ps1:66]: Email shared.

File: tools/TestFx/Live/SaveLiveTestResult.ps1 (hunk 2 above)
Comment (vidai-msft) [reply tools/TestFx/Live/SaveLiveTestResult.ps1:28]: @isra-fel Publish is still not that accurate. How about **Save-KustoDataFromCsv**? Or just **Save-KustoData**. Later on, this cmdlet can expand to save the data from multiple sources like json, storage via different parametersets. Your thoughts?

File: tools/TestFx/Live/InvokeLiveTestScenarios.ps1 (hunk 1 above)
Comment (vidai-msft) [reply tools/TestFx/Live/InvokeLiveTestScenarios.ps1]: Comment added.

File: tools/TestFx/Live/SaveLiveTestResult.ps1 (hunk 2 above)
Comment (isra-fel) [reply tools/TestFx/Live/SaveLiveTestResult.ps1:28]: Let's use `Add-KustoData`

**Review by isra-fel — APPROVED** [review]

**Pull request merged**